Network segments docs enhancements (#10702)
hashicorp/consul: website/content/docs/enterprise/network-segments.mdx

<EnterpriseAlert>
  This feature requires{' '}
  <a href="https://www.hashicorp.com/products/consul/">Consul Enterprise</a>{' '}
  version 0.9.3+ with the Global Visibility, Routing, and Scale module.
</EnterpriseAlert>

Consul requires full connectivity between all agents (servers and clients) in a

> datacenter within a given LAN gossip pool. By default, all Consul agents will be

Author: @blake : do we refer to this as a "datacenter" or a "cluster" now? I ask because under the Consul Networking Models section, we use "Cluster". Interestingly, Cluster isn't a recognized term in the [glossary](https://www.consul.io/docs/install/glossary#datacenter), but Datacenter is.

Reviewer: Agents are configured to be members of a specific [datacenter](https://www.consul.io/docs/agent/options#_datacenter). By default, within a given datacenter, Consul requires full LAN gossip connectivity between all agents.

a part of one shared LAN gossip pool known as the `<default>` network segment,
thus requiring full mesh connectivity within the datacenter.

![Consul datacenter default agent connectivity: one network segment](/img/network-segments/consul-network-segments-single.png)

But many advanced Consul users have the need to run a datacenter with a segmented

> network, meaning that there is not full connectivity between all agents. This is

Author: Here's my attempt at triangulating the feedback on this section both here and on the Learn tutorial PR:

```
In some environments, full connectivity between all agents is not possible—known as a
"segmented network". This is usually the result of business policies enforced through
network rules or firewalls. To use Consul in a segmented network, you must break up
the LAN gossip pool along network segment boundaries into "network segments". Network
segments are isolated LAN gossip pools with full connectivity between all agent members.
```

usually the result of business policies enforced via network rules or firewalls.
To use Consul with a segmented network, break up the LAN gossip pool along
network segment boundaries into "network segments": isolated LAN gossip pools

> with full connectivity between all agent members.

Author: What does `unique` mean in the context of `unique "network segments"`?

> with full connectivity between all agent members.

Reviewer: "Separate" might be a better word.

Using the word "segment" in two different contexts is confusing. Ultimately this should be worded in way that a Consul network segments should be used when there is restrictive / no communication between nodes across L3 network boundaries within the data center.

> with full connectivity between all agent members.

Author: Changed to:
```
To use Consul in a segmented network, you must break up
the LAN gossip pool along network communication boundaries into separate "network
segments". Network segments are isolated LAN gossip pools with full connectivity
between all agent members.
```

![Consul datacenter agent connectivity with network segments](/img/network-segments/consul-network-segments-multiple.png)

To get started with network segments you can review the tutorial on HashiCorp
Learn for [Network Segments](https://learn.hashicorp.com/tutorials/consul/network-partition-datacenters).

> -> **Info:** Network segments enable operating a datacenter without full mesh

Author: I agree with changing `operating` to `you to operate`, but I find the meaning less clear if I switch `without full mesh connectivity` to `with limited mesh connectivity`.

> -> **Info:** Network segments enable operating a datacenter without full mesh

Reviewer 2: It's your call. For many people, concepts can be easier to consume when they are described in terms of presence -- _having_ something instead of _not having_ something. That's why I suggested the 'without => with' change. Perhaps 'limited' isn't concrete enough. I defer to you.

> -> **Info:** Network segments enable operating a datacenter without full mesh

Author: I generally agree. In this case, my understanding is that "mesh" = "everything is connected", so "limited mesh" isn't a clear phrase. I'll spend a few minutes seeing if I can switch the statement to a "having something" statement with different wording. If not, I'll keep as is.

(LAN) connectivity between agents. To enable operating multiple federated Consul
datacenters without full mesh (WAN) connectivity between server agents, use
[network areas (Enterprise)](https://www.consul.io/api-docs/operator/area) instead.

> ## Server Configuration

Author: @blake : I'm declining this (and similar) suggestions.

Per the [HashiCorp Voice, Style, and Language guidelines](https://docs.google.com/document/d/1MRvGd6tS5JkIwl_GssbyExkMJqOXKeUE00kSEtFi8m8/edit#) (CTRL+F for "Titles and subheadings"), titles and subheadings should use title case.

Server agents are members of all segments and therefore provide a connection
point across segments. Segments include (1) the `<default>` segment and (2) the

> additional segments specified in their [`segments`](https://www.consul.io/docs/agent/options.html#segments)

Reviewer 2: See this comment, too:
https://github.com/hashicorp/learn/pull/3712/files#r681169777

agent configuration option. Each additional segment specifies a non-empty name
and unique port, and optionally a custom bind and advertise address, for that
segment's listener on the server.

> ~> **Note:** Prior to Consul 1.7.3, a Consul server agent configured with too

Author: I've decided not to take this suggestion.

Per the engineering docs style guide (that you recently asked the education team about on Slack):

The blue box (`->`) is for: Tips, tricks, and recommended practices
The yellow box (`~>`) is for: Notes, concerns, and potential pitfalls

I think this is more of a concern / potential pitfall (yellow box).

many network segments may not be able to start due to [limitations](https://learn.hashicorp.com/tutorials/consul/network-partition-datacenters#network-segments-limitations)
in Serf.

### Example Server Configuration

If the snippet below is added to a server agent configuration file, it will
create two user-defined segments: `alpha` and `beta`.

<Tabs>
<Tab heading="HCL">

```hcl
segments= [
  {
    name= "alpha"
    bind= "{{GetPrivateIP}}"
    advertise= "{{GetPrivateIP}}"
    port= 8303
  },
  {
    name= "beta"
    bind= "{{GetPrivateIP}}"
    advertise= "{{GetPrivateIP}}"
    port= 8304
  }
]
```

</Tab>
<Tab heading="JSON">

```json
{
  "segments": [
    {
      "name": "alpha",
      "bind": "{{GetPrivateIP}}",
      "advertise": "{{GetPrivateIP}}",
      "port": 8303
    },
    {
      "name": "beta",
      "bind": "{{GetPrivateIP}}",
      "advertise": "{{GetPrivateIP}}",
      "port": 8304
    }
  ]
}
```

</Tab>
</Tabs>

> List of relevant server configuration parameters, all within the server

Reviewer 2: ```suggestion
The following parameters are relevant configurations within the server
```

Author: Changed to

```
The server [agent configuration](/docs/agent/options) options relevant to network
segments are:
```

[agent configuration](https://www.consul.io/docs/agent/options):

- [`ports.serf_lan`](https://www.consul.io/docs/agent/options#serf_lan_port):
  listener port for `<default>` segment on this server
- [`segments`](https://www.consul.io/docs/agent/options#segments): for

>   configuring user-defined segments on this server

Author: @blake : the intent was to provide a list of all configuration parameters which may be relevant to network segments. I don't want to just refer to the agent configuration page in general because it may be very difficult to pull out the relevant parameters from there.

As a reader, what was confusing / do you have any suggestions on what to change?

> List of relevant server configuration parameters, all within the server
> [agent configuration](https://www.consul.io/docs/agent/options):
> 
> - [`ports.serf_lan`](https://www.consul.io/docs/agent/options#serf_lan_port):
>   listener port for `<default>` segment on this server
> - [`segments`](https://www.consul.io/docs/agent/options#segments): for
>   configuring user-defined segments on this server

Reviewer: Its not clear what the intent is from the current phrasing. This, for example, might be more clear.

```suggestion
Below are a list of additional agent configuration options which are relevant to network segments:
[agent configuration](https://www.consul.io/docs/agent/options):

- [`ports.serf_lan`](https://www.consul.io/docs/agent/options#serf_lan_port):
  The Serf LAN port for `default` gossip pool on the server.
- [`segments`](https://www.consul.io/docs/agent/options#segments): A list of
   user-defined network segments on this server.
```

## Client Configuration

Client agents are members of only one segment, which will be the `<default>`
segment unless otherwise specified in their [`segment`](https://www.consul.io/docs/agent/options.html#_segment)

> agent configuration option.

Author: Changed to

```
Each client agent is a member of only one segment at a time. This will be the
`<default>` segment unless otherwise specified in their [`segment`](/docs/agent/options.html#_segment)
agent configuration option.
```

### Join a Client to a Segment ((#join_a_client_to_a_segment))

For a client agent to [join](https://www.consul.io/commands/join) the Consul
datacenter, it must connect to another agent (client or server) within its
configured segment.

<Tabs>
<Tab heading="Join via Another Client">

Clients A and B specify the same segment S. Client B is already in the
datacenter. Client A wants to join via Client B. To succeed, Client A's join
operation must connect to Client B's [LAN listener port](https://www.consul.io/docs/agent/options#serf_lan_port).

</Tab>
<Tab heading="Join via a Server">

Client A specifies segment S and wants to join via Server 1. To succeed, Client

> A's join operation must connect to Server 1's [LAN listener port for segment S](https://www.consul.io/docs/agent/options#segment_port).

Author: Why "segment port for segment S" instead of "LAN listener port for segment S"?

> A's join operation must connect to Server 1's [LAN listener port for segment S](https://www.consul.io/docs/agent/options#segment_port).

Reviewer: The port is referred to in the documentation as being the segment port, not a LAN listener port.

> A's join operation must connect to Server 1's [LAN listener port for segment S](https://www.consul.io/docs/agent/options#segment_port).

Author: Changed to: `LAN port for segment S`. Within the context of the network segments documentation, I think it makes it easier to understand that there's no distinction between "LAN ports" and "segment ports"---they are all LAN ports for segments (`<default>` or user-defined).

</Tab>
</Tabs>

There are several ways to specify the port used by the join operation, in order
of precedence:

1. An explicit port in the join address, if specified. For example, specified as
   port 8303 from the CLI with `consul agent -retry-join "client-b-address:8303"`,
   or from Client A's configuration with the [retry join option](https://www.consul.io/docs/agent/options#retry_join).
2. Client A's [LAN listener port](https://www.consul.io/docs/agent/options#serf_lan_port),
   if specified. For example, specified as port 8303 from the CLI with
   `consul agent -retry-join "client-b-address" -serf-lan-port 8303`,

>    or from Client A's agent configuration with the [LAN listener port option](https://www.consul.io/docs/agent/options#serf_lan_port)

Author: Changed to:

```
Specify a LAN port for Client A. Without an explicit LAN port in the join address,
Client A will implicitly use its own port number. For example,
`consul agent -retry-join "client-b-address" -serf-lan-port 8303` when starting Client A
using the CLI, or in Client A's configuration using the
[serf_lan](/docs/agent/options#serf_lan_port) option.
```

3. The default LAN listener port (8301).

Agents within a segment can use different port numbers for their LAN listener.
For example, on the `<default>` segment, Server S can use port 8301, Client A
can use 8303, and Client B can use 8304. But if an agent wishes to join via an
agent that uses a different port number, that target port must be specified in

> the join address.

Author: Made most of these changes, but not all

### Example Client Configuration

If the snippet below is added to a client agent configuration file, the client
will:

- Open a LAN listener port on 8303
- Attempt to join the cluster via servers on port 8303 (since no port is
  explicitly defined in the `retry_join` addresses)

<Tabs>
<Tab heading="HCL">

```hcl
"node_name" = "consul-client1"
"retry_join" = ["consul-server1", "consul-server2", "consul-server3"]
"segment" = "alpha"
"ports" = {
  "serf_lan" = 8303
}
```

</Tab>
<Tab heading="JSON">

```json
{
  "node_name": "consul-client1",
  "retry_join": ["consul-server1", "consul-server2", "consul-server3"],
  "segment": "alpha",
  "ports": {
    "serf_lan": 8303
  }
}
```

</Tab>
</Tabs>

List of relevant client configuration parameters, all within the client [agent configuration](https://www.consul.io/docs/agent/options):

> - [`segment`](https://www.consul.io/docs/agent/options#segment-2)
> - [`ports.serf_lan`](https://www.consul.io/docs/agent/options#serf_lan_port):
>   listener port for the above segment on this client; does not need to be the
>   same as the listener port for other agents on this segment
> - [`retry_join`](https://www.consul.io/docs/agent/options#retry_join) or
>   [`start_join`](https://www.consul.io/docs/agent/options#start_join):
>   ensure the correct LAN listener port for this segment on the join target is
>   used, whether [by explicitly including a port in the join address or by other
>   available means](#join_a_client_to_a_segment)

Reviewer: ```suggestion
- [`segment`](/docs/agent/options#segment-2)
- [`ports.serf_lan`](/docs/agent/options#serf_lan_port):
  Listener port for the above segment on this client. This is not required to match the configured listener port for other agents on this segment.
- [`retry_join`](/docs/agent/options#retry_join) or
  [`start_join`](/docs/agent/options#start_join):
  Ensure the correct Serf LAN port for this segment is
  used when joining the gossip pool by specifying the port using one of the
  [available configuration methods](#join_a_client_to_a_segment).
```

<EnterpriseAlert>
  This feature requires{' '}
  <a href="https://www.hashicorp.com/products/consul/">Consul Enterprise</a>{' '}
  version 0.9.3+ with the Global Visibility, Routing, and Scale module.
</EnterpriseAlert>

Consul requires full connectivity between all agents (servers and clients) in a
[datacenter](https://www.consul.io/docs/agent/options#_datacenter) within a given
LAN gossip pool. By default, all Consul agents will be a part of one shared Serf LAN
gossip pool known as the `<default>` network segment, thus requiring full mesh
connectivity within the datacenter.

![Consul datacenter default agent connectivity: one network segment](/img/network-segments/consul-network-segments-single.png)

In some environments, full connectivity between all agents is not possible—known as a

> "segmented network". This is usually the result of business policies enforced through

Author: It actually is an em dash, it just renders in the text file in such a way that it appears like a hyphen!

(Note to self: double check it appears as an em dash in the rendered version.)

> "segmented network". This is usually the result of business policies enforced through

Author: ![image](https://user-images.githubusercontent.com/85913323/129201891-b08920c3-e132-4b65-a0eb-d6638ec52dc9.png)

network rules or firewalls. To use Consul in a segmented network, you must break up
the LAN gossip pool along network communication boundaries into separate "network
segments". Network segments are isolated LAN gossip pools with full connectivity

> between all agent members.

Author: Changed to:
```
Network segments are isolated LAN gossip pools which require full connectivity
only between all agent members on the same segment.
```

agent configuration option.

### Join a Client to a Segment ((#join_a_client_to_a_segment))

For a client agent to [join](/commands/join) the Consul
datacenter, it must connect to another agent (client or server) within its
configured segment.

<Tabs>
<Tab heading="Join via another client">

Clients A and B specify the same segment S. Client B is already joined to the segment S
LAN gossip pool. Client A wants to join via Client B. In order to do so, Client A
must connect to Client B's configured [Serf LAN port](/docs/agent/options#serf_lan_port).

</Tab>
<Tab heading="Join via a server">

Client A specifies segment S and wants to join the segment S gossip pool via Server 1.
In order to do so, Client A must connect to Server 1's configured [Serf LAN port
for segment S](/docs/agent/options#segment_port).

</Tab>
</Tabs>

There are several methods to specify the port used by the join operation, listed in order
of precedence:

1. Specify an explicit port in the join address. For example,
   `consul agent -retry-join "client-b-address:8303"` when starting Client A using the CLI,
   or in Client A's configuration using the [retry-join option](/docs/agent/options#retry_join)

>    without cloud auto-join. If using `retry-join`

Author: @blake : FYI - the clarification in here about `retry-join` with cloud auto-join is completely new since your last review.

> 1. Specify an explicit port in the join address. For example,
>    `consul agent -retry-join "client-b-address:8303"` when starting Client A using the CLI,
>    or in Client A's configuration using the [retry-join option](/docs/agent/options#retry_join)
>    without cloud auto-join. If using `retry-join`
>    [with cloud auto-join](https://www.consul.io/docs/install/cloud-auto-join#auto-join-with-network-segments),
>    there is no method to specify an explicit port to use, so one of the implicit methods
>    below must be used instead.

Reviewer: This gives the impression that cloud auto-join might be usable when the segment port is defined at the CLI.

```suggestion
1. Specify an explicit port in the join address. For example,
   `consul agent -retry-join "client-b-address:8303"` when starting the agent using the CLI,
   or in the agent's configuration using the [retry-join option](/docs/agent/options#retry_join). This method is not compatible with [cloud auto-join](/docs/install/cloud-auto-join#auto-join-with-network-segments).
```

> 2. Specify a Serf LAN port for Client A. Without an explicit Serf LAN port in the join address,
>    Client A will implicitly use its own port number. For example,
>    `consul agent -retry-join "client-b-address" -serf-lan-port 8303` when starting Client A
>    using the CLI, or in Client A's configuration using the
>    [serf_lan](/docs/agent/options#serf_lan_port) option.

Reviewer: ```suggestion
2. Specify an alternate Serf LAN port at the CLI using the `-serf-lan-port` flag (e.g., `consul agent -retry-join "client-b-address" -serf-lan-port 8303`), or in the agent's configuration using the [`ports.serf_lan`] option. When a Serf LAN port is not explicitly specified in the join address, the agent will join the Serf LAN port specified at the CLI or agent configuration.
```

Author: Thanks for the cleanup. I made a slightly different set of changes (mostly to the last part). Let me know if you agree:

```
2. Specify an alternate Serf LAN port for the agent. This can be done at the CLI when starting
   the agent (e.g., `consul agent -retry-join "client-b-address" -serf-lan-port 8303`), or in
   the agent's configuration using the [serf_lan](/docs/agent/options#serf_lan_port) option.
   When a Serf LAN port is not explicitly specified in the join address, the agent will use its
   own Serf LAN port number as the join target port. This method requires that the agent and its
   join target use the same Serf LAN port for this segment.
```

Reviewer: I find this to be confusing.

> The agent will use its own Serf LAN port number as the join target port. This method requires that the agent and its join target use the same Serf LAN port for this segment.

"Its own port number" implies a local port to me. The phrasing is ambiguous, and could be interpreted that the agent will contact itself when in actuality the agent will initiate a connection a port on a remote host.

This is probably more clear as to the actual operation.

> When a Serf LAN port is not explicitly specified in the join address, the agent will attempt to join the specified host at the Serf LAN port specified in CLI or agent configuration.

Author: We don't use the word host anywhere else in the guide. Can I use `target agent` instead of `target host`?

Author: Used your wording for the last part of item 2.

Also modified item 3 to include similar text:
```
3. **Use the default Serf LAN port (8301)**. The agent will attempt to join the target host
   on port 8301.
```

3. The default Serf LAN port (8301).

-> Agents within a segment can use different port numbers for their Serf LAN port.
For example, on the `<default>` segment, Server S can use port 8301, Client A
can use 8303, and Client B can use 8304. However, if an agent wishes to join a
segment via an agent that uses a different port number, the target agent's Serf LAN
port must be specified in the join address (option 1 above).

### Example Client Configuration

The following client agent configuration will cause the agent to:

- Open a Serf LAN listener port on 8303.
- Attempt to join the cluster via servers on port 8303 (since an alternative port is not
  specified in the `retry_join` addresses).

<CodeTabs heading="Example network segment configuration for client agents">

```hcl

> "node_name" = "consul-client1"
> "retry_join" = ["consul-server1", "consul-server2", "consul-server3"]
> "segment" = "alpha"
> "ports" = {
>   "serf_lan" = 8303
> }

Reviewer: ```suggestion
node_name = "consul-client1"
retry_join = ["consul-server1", "consul-server2", "consul-server3"]
segment = "alpha"
ports = {
  serf_lan = 8303
}
```

```

```json
{
  "node_name": "consul-client1",
  "retry_join": ["consul-server1", "consul-server2", "consul-server3"],
  "segment": "alpha",
  "ports": {
    "serf_lan": 8303
  }
}
```

</CodeTabs>

The client [agent configuration](/docs/agent/options) options relevant to network
segments are:

- [`segment`](/docs/agent/options#segment-2): The name of the network segment this
  client agent belongs to.
- [`ports.serf_lan`](/docs/agent/options#serf_lan_port):
  Serf LAN port for the above segment on this client. This is not required
  to match the configured Serf LAN port for other agents on this segment.
- [`retry_join`](/docs/agent/options#retry_join) or
  [`start_join`](/docs/agent/options#start_join): A list of agent addresses to join
  when starting. Ensure the correct Serf LAN port for this segment is used when joining
  the LAN gossip pool using one of the [available configuration methods](#join_a_client_to_a_segment).

> # Consul Networking Models

Author: @blake : I _think_ the last remaining question I have is where this Core Networking Models section belongs.

You previously said:
> The tutorial on the network segments page should probably come after the Consul Networking Models section so the user has the proper context on the networking models before reading how to configure the feature.

> Another option would be to move the tutorial be moved to a different page entirely and referenced from this main page. Although, at that point it may duplicate what is on the Learn guide. 🤔

I view the Learn guide as a "How To" whereas this is page is more of a complete reference. It could possibly make sense to separate a Network Segments overview from the details, though I don't think this page is soooo long as to make that necessary. I'm thinking we leave it all on this one page for now.

I think the first section on this page should definitely be the Network Segments overview that is shown right now (with the images), not the Core Networking Models. 

Perhaps it could look like the below. Do you agree?
```
# Network Segments
// ... the overview explanation with the images ...

# Consul Networking Models
// ...

# Configuring Network Segments

## Server Configuration
### Example Server Configuration
// ... the rest of the new content
```

Author: @blake : I'll go through your last round of comments (from Aug 23) soon, but before I do, I want to see if you have an opinion on this (where the Core Networking Models section belongs). I'd like to batch all the remaining changes into one edit if possible.

<EnterpriseAlert>
  This feature requires{' '}
  <a href="https://www.hashicorp.com/products/consul/">Consul Enterprise</a>{' '}
  version 0.9.3+ with the Global Visibility, Routing, and Scale module.
</EnterpriseAlert>

Consul requires full connectivity between all agents (servers and clients) in a
[datacenter](/docs/agent/options#_datacenter) within a given
LAN gossip pool. By default, all Consul agents will be a part of one shared Serf LAN
gossip pool known as the `<default>` network segment, thus requiring full mesh
connectivity within the datacenter.

![Consul datacenter default agent connectivity: one network segment](/img/network-segments/consul-network-segments-single.png)

In some environments, full connectivity between all agents is not possible—known as a
"segmented network". This is usually the result of business policies enforced through
network rules or firewalls. To use Consul in a segmented network, you must break up
the LAN gossip pool along network communication boundaries into separate "network

> segments". Network segments are isolated LAN gossip pools which require full connectivity
> only between all agent members on the same segment.

Reviewer: ```suggestion
segments". Network segments are isolated LAN gossip pools that only require full connectivity between agents on the same segment.
```

![Consul datacenter agent connectivity with network segments](/img/network-segments/consul-network-segments-multiple.png)

To get started with network segments you can review the tutorial on HashiCorp
Learn for [Network Segments](https://learn.hashicorp.com/tutorials/consul/network-partition-datacenters).

> -> **Info:** Network segments enable you to operate a datacenter without full mesh
> (LAN) connectivity between all agents. To enable operating multiple federated Consul
> datacenters without full mesh (WAN) connectivity between all server agents, use
> [Network Areas (Enterprise)](/docs/enterprise/federation) instead.

Reviewer: ```suggestion
-> **Info:** Network segments enable you to operate a Consul datacenter without full mesh
(LAN) connectivity between agents. To federate multiple Consul
datacenters without full mesh (WAN) connectivity between datacenters, use
[Network Areas (Enterprise)](/docs/enterprise/federation).
```
Within a given area, servers between peered DCs are fully meshed.

Author: Instead of `without full mesh (WAN) connectivity between datacenters`, should it be `... between Consul servers`? The first phrasing implies (to me) that we require full mesh connectivity between all agents across federated datacenters (which I didn't think was true), not just across all server agents.

Reviewer: > The first phrasing implies (to me) that we require full mesh connectivity between all agents across federated datacenters.

WAN federation only takes place between servers, not clients.

By default, federated datacenters form a full mesh of connectivity to servers in other DCs on the WAN gossip pool. Network Areas are useful when it is not possible, or not desirable, to have a full mesh of connectivity between all servers in all datacenters.

Within an area, servers establish a full mesh of connectivity to servers in the peer DC. As such, it's correct to say that this feature is useful when there is not full mesh connectivity between all _DCs_, as all servers within the area still require the ability to directly communicate to their peers.

Author: > As such, it's correct to say that this feature is useful when there is not full mesh connectivity between all DCs, as all servers within the area still require the ability to directly communicate to their peers.

We're on the same page. Network segments are to agents within a DC as network areas are to server agents across DCs.

I'm just trying to figure out how to embed that concisely in the sentence. If we assume someone already understands clearly how Consul WAN federation works, then I agree "between all datacenters" will be understood. Is that an assumption we want to make here? And if not, is there something else we can say?

e.g., `without full mesh (WAN) connectivity between all server agents in all datacenters` instead of just `between all datacenters`?

Author: Changed to:
```
To federate multiple Consul datacenters
without full mesh (WAN) connectivity between all server agents in all datacenters,
```

pool for different purposes, like finding a Consul server in a local cluster,
or finding servers in a remote cluster. A **segmented** LAN Gossip Pool limits a
group of agents to only connect with the agents in its segment.

# Network Segments Configuration

## Server Configuration

> Server agents are members of all segments and, therefore, provide a connection
> point across segments. The datacenter includes a `<default>` segment, as well as

Reviewer: ```suggestion
Server agents are members of all segments. The datacenter includes a `<default>` segment, as well as
```
This might be misleading as the servers do not facilitate routing gossip traffic between segments.

segments: `alpha` and `beta`.

<CodeTabs heading="Example network segments configuration for server agents">

```hcl
segments = [
  {
    name = "alpha"
    bind = "{{GetPrivateIP}}"
    advertise = "{{GetPrivateIP}}"
    port = 8303
  },
  {
    name = "beta"
    bind = "{{GetPrivateIP}}"
    advertise = "{{GetPrivateIP}}"
    port = 8304
  }
]
```

```json
{
  "segments": [
    {
      "name": "alpha",
      "bind": "{{GetPrivateIP}}",
      "advertise": "{{GetPrivateIP}}",
      "port": 8303
    },
    {
      "name": "beta",
      "bind": "{{GetPrivateIP}}",
      "advertise": "{{GetPrivateIP}}",
      "port": 8304
    }
  ]
}
```

</CodeTabs>

The server [agent configuration](/docs/agent/options) options relevant to network
segments are:

- [`ports.serf_lan`](/docs/agent/options#serf_lan_port): The Serf LAN port on this server
  for the `<default>` network segment's gossip pool.
- [`segments`](/docs/agent/options#segments): A list of user-defined network segments
  on this server, including their names and Serf LAN ports.

## Client Configuration

Each client agent can only be a member of one segment at a time. This will be the

> `<default>` segment unless otherwise specified in their [`segment`](/docs/agent/options.html#_segment)
> agent configuration option.

Reviewer: ```suggestion
`<default>` segment unless otherwise specified in the agent's [`segment`](/docs/agent/options.html#_segment) configuration option.
```
